sdk/typescript/src/builder/Transaction.ts

```diff
@@ -185,23 +192,40 @@ export class Transaction {
    * is the format required for custom serialization.
    *
    */
-  input(value?: unknown) {
+  #input(type: 'object' | 'pure', value?: unknown) {
     // For Uint8Array
     // if (value instanceof Uint8Array) {
     //   value = { Pure: value };
     // }
 
     const index = this.#transactionData.inputs.length;
-    const input = create({ kind: 'Input', value, index }, TransactionInput);
+    const input = create(
+      { kind: 'Input', value, index, type },
+      TransactionInput,
+    );
     this.#transactionData.inputs.push(input);
     return input;
   }
 
-  // TODO: Do we want to support these helper functions for inputs?
-  // Maybe we can make an `Inputs` helper like commands that works seamlessly with these.
-  // objectRef() {}
-  // sharedObjectRef() {}
-  // pure() {}
+  /**
+   * Add a new object input to the transaction.
+   */
+  object(value: ObjectId | ObjectCallArg) {
```

Comment:
without user-provided hints about `object` and `pure`, we can't dedupe at this stage due to edge cases where a transaction might have an object id input and a pure input that are exactly the same.